site/docs/v1/tech/admin-guide/upgrade.adoc

```diff
@@ -7,41 +7,270 @@ navcategory: admin
+* You have concerns about compatibility with customizations or third-party integrations.
+
+=== Being Aware of Upgrade Options
+
+You can stay informed about new FusionAuth releases and updates by:
+
+* Monitoring the link:/docs/v1/tech/release-notes[release notes].
+* Joining the https://fusionauth.io/community[FusionAuth Community site] and https://fusionauth.slack.com/join/shared_invite/zt-6ykfov9u-MTwA1A6ptv0Gr6KcFBVG4w#/shared-invite/email[Slack channel].
+* Following FusionAuth on social media platforms, including https://twitter.com/FusionAuth[Twitter] and https://www.linkedin.com/company/fusionauth[LinkedIn].
+* Monitoring the https://github.com/FusionAuth/[FusionAuth GitHub repository].
+
+== How to Think About Upgrades
+
+Before upgrading FusionAuth, there are a number of things to consider. Go through each of the points below to see if any of them apply to your situation.
+
+=== Update Contents and Notes
+
+Read the link:/docs/v1/tech/release-notes[release notes]. Release notes help you understand how an update affects you. They include details on new features, improvements, bug fixes, and potential breaking changes.
+
+=== Testing
+
+Before starting the upgrade process on production, test the upgrade in a staging or development environment to identify potential issues. This is particularly important on more complex upgrades involving multiple version jumps or if you have made customizations to FusionAuth.
+
+=== Downtime and Database Migrations
+
+Plan for any downtime or maintenance windows required for the upgrade. If changes to the database schema are required, this could require more steps and downtime that affects end users. A pure application upgrade, on the other hand, may only affect admin functionality and not require any downtime from an end-user perspective.
+
+FusionAuth can perform database migrations automatically and silently using a process called link:/docs/v1/tech/guides/silent-mode[Silent Mode]. 
+
+If you want to run the migrations interactively, use link:/docs/v1/tech/installation-guide/fusionauth-app#maintenance-mode[Maintenance Mode], or you can run any needed scripts out of band (meaning directly on the database yourself).
+
+[NOTE]
+====
+You should always back up your database before using Maintenance Mode or performing any database migrations by any method.
+====
+
+The timing of the database upgrade is also important to note. When an upgraded node is added to a FusionAuth cluster, it will lock the database and perform the database upgrade if configured for Silent Mode. This means that the database will be unavailable to other nodes in the cluster until the upgrade is complete. 
+
+=== Connected Applications and Client Libraries
+
+Consider that a FusionAuth server upgrade might mean that you need to upgrade your client libraries too. Check if this is needed before upgrading, as it will entail a more complex upgrade process and may require more downtime.
```

Comment:
```suggestion
Consider that a FusionAuth server upgrade might mean that you need to upgrade your client libraries too. Check if this is needed before upgrading, as it will entail a more complex upgrade process and may require more downtime.

In general, client libraries are forward compatible, but any issues will be noted in the release notes. (Read the release notes carefully.)
```